### PR TITLE
Optimise dockerfile

### PR DESCRIPTION
**Pull request by modem7**
Reduced the number of layers from 26 to 18 resulting in faster download, slightly faster build (due to layer caching), and negligible smaller size.

Added python version variable to decrease maintenance (also allows for quick testing of different python versions during builds as can declare it as a build arg).

Added software version to entrypoint script so users can more easily provide version numbers in issues.

This has been tested locally.

### CODE REVIEW

**Comment by modem7** [comment]
Please note, no changes were made to the msmtp scripts. This is an unfortunate side effect of having my local github files in Linux, whilst modifying them in Windows.

**Comment by toastie89** [comment]
Hi @modem7, looks like this change introduces two more packages (zstd, libxxhash), may I ask for what they are?

File: base/Dockerfile [hunk 1]
```diff
@@ -31,42 +31,50 @@ RUN python3 -m pip install -v --no-cache-dir -U -r requirements.txt
...
+RUN apk add --update --no-cache \
     tzdata \
     sshfs \
     python3 \
     openssl \
+    libxxhash \
+    zstd \
     fuse \
     ca-certificates \
+    logrotate \
```
Comment (grantbevis) [inline base/Dockerfile:52]: Also why are we installing logrotate if we're ditching the msmtp version?

File: base/Dockerfile [hunk 2]
```diff
@@ -31,42 +31,50 @@ RUN python3 -m pip install -v --no-cache-dir -U -r requirements.txt
...
-    && apk add --no-cache \
+ARG PYTHON_VERSION=3.10
+VOLUME /mnt/source
+VOLUME /mnt/borg-repository
+VOLUME /root/.borgmatic
+VOLUME /etc/borgmatic.d
+VOLUME /root/.config/borg
+VOLUME /root/.ssh
+VOLUME /root/.cache/borg
+HEALTHCHECK --interval=30s --timeout=10s --start-period=20s --retries=3 CMD borgmatic --version || exit 1
```
Comment (grantbevis) [inline base/Dockerfile:42]: Is this really necessary? Seems pointless to me

File: base/Dockerfile [hunk 3]
```diff
@@ -31,42 +31,50 @@ RUN python3 -m pip install -v --no-cache-dir -U -r requirements.txt
...
+                    /usr/bin/borg \
+                    /usr/bin/borgfs \
+                    /usr/bin/borgmatic \
+                    /usr/bin/generate-borgmatic-config \
+                    /usr/bin/upgrade-borgmatic-config \
+                    /usr/bin/validate-borgmatic-config \
+                    /usr/bin/markdown_py \
+                    /usr/bin/normalizer \
+                    /usr/bin/apprise \
+                    /usr/bin/
```
Comment (grantbevis) [inline base/Dockerfile:77]: Dodgy indentation here bud, the first / should roughly line up with --chmod

**Review by grantbevis — CHANGES_REQUESTED** [review]
Please see other comments

File: base/Dockerfile (hunk 2 above)
Comment (modem7) [reply base/Dockerfile:42]: Not a requirement. Can be removed if there's no value-add.

File: base/Dockerfile (hunk 3 above)
Comment (modem7) [reply base/Dockerfile]: Rgr that! Resolved.

**Comment by modem7** [comment]
> Hi @modem7, looks like this change introduces two more packages (zstd, libxxhash), may I ask for what they are?

Heya, 

I found that with this way of building (via pre-built wheels), borg refused to work properly without these being added on alpine 3.16.x. 

Unfortunately, I don't know enough about python to debug without assistance. I'm happy to assist with debugging.

**Comment by grantbevis** [comment]
> It may be worthwhile for me to recreate the pull request with a fresh branch, so we can have a cleaner look. I'll wait until @grantbevis replies regarding the indentation and healthcheck prior to me doing so.

Can do bud, maybe we should adopt having a `develop` branch

**Comment by modem7** [comment]
> > It may be worthwhile for me to recreate the pull request with a fresh branch, so we can have a cleaner look. I'll wait until @grantbevis replies regarding the indentation and healthcheck prior to me doing so.
> 
> Can do bud, maybe we should adopt having a `develop` branch

Aight, I'll create a develop branch and move my changes across to there so we can have a fresh set of eyes due to my error above. 

I'll close this PR and raise a new one and give you guys a ping.

**Pull request closed without merging**